db/crud_test.go

```diff
@@ -956,3 +956,219 @@ func BenchmarkHandleRevDelta(b *testing.B) {
+	doc, rev1, err := db.PutExistingRevWithBody(docId, unjson(payload), []string{"1-a"}, false)
+	assert.NoError(t, err, "Couldn't create document")
+	assert.NotEmpty(t, doc, "Document shouldn't be empty")
+	assert.Equal(t, "1-a", rev1, "Provided input revision ID should be returned")
+
+	// Get the 1x revision with rev1
+	bodyBytes, removed, err := db.get1xRevFromDoc(doc, rev1, true)
+	assert.NotEmpty(t, bodyBytes, "Document body bytes should be returned")
+	assert.False(t, removed, "This shouldn't be a removed document")
+	var response = Body{}
+	assert.NoError(t, response.Unmarshal(bodyBytes))
+	assert.Equal(t, docId, response[BodyId])
+	assert.Equal(t, "1-a", response[BodyRev])
+	assert.Equal(t, "Los Angeles", response["city"])
+
+	// Create the second revision of the document
+	payload = `{"city":"Hollywood"}`
+	doc, rev2, err := db.PutExistingRevWithBody(docId, unjson(payload), []string{"2-a", "1-a"}, false)
+	assert.NoError(t, err, "Couldn't create document")
+	assert.NotEmpty(t, doc, "Document shouldn't be empty")
+	assert.Equal(t, "2-a", rev2, "Provided input revision ID should be returned")
+
+	// Get the 1x revision with rev2
+	bodyBytes, removed, err = db.get1xRevFromDoc(doc, rev2, true)
+	assert.NotEmpty(t, bodyBytes, "Document body bytes should be returned")
+	assert.False(t, removed, "This shouldn't be a removed document")
+	assert.NoError(t, response.Unmarshal(bodyBytes))
+	assert.Equal(t, docId, response[BodyId])
+	assert.Equal(t, "2-a", response[BodyRev])
+	assert.Equal(t, "Hollywood", response["city"])
+
+	// Get body bytes from doc with unknown revision id; it simulates the error scenario.
+	// A 404 missing error should be thrown when trying get the body bytes of the document
+	// which doesn't exists in the database. The revision "3-a" doesn't exists in database.
+	bodyBytes, removed, err = db.get1xRevFromDoc(doc, "3-a", true)
+	assert.Error(t, err, "It should throw 404 missing error")
+	assert.Contains(t, err.Error(), "404 missing")
+	assert.Empty(t, bodyBytes, "Provided revision doesn't exists")
+	assert.False(t, removed, "This shouldn't be a removed revision")
+	assert.Error(t, response.Unmarshal(bodyBytes), "Unexpected empty JSON input to body.Unmarshal")
+
+	// Deletes a document, by adding a new revision whose _deleted property is true.
+	body := Body{BodyDeleted: true, BodyRev: rev2}
+	rev3, doc, err := db.Put(docId, body)
+	assert.NoError(t, err, "Document should be deleted")
+	assert.NotEmpty(t, rev3, "Document revision shouldn't be empty")
+
+	// Get the document body bytes with the deleted revision, list revisions enabled.
+	bodyBytes, removed, err = db.get1xRevFromDoc(doc, rev3, true)
+	assert.NotEmpty(t, bodyBytes, "Document body bytes should be returned")
+	assert.False(t, removed, "This shouldn't be a removed document")
+	assert.NoError(t, response.Unmarshal(bodyBytes))
+	assert.Equal(t, docId, response[BodyId])
+	assert.Equal(t, rev3, response[BodyRev])
+	assert.Equal(t, "Hollywood", response["city"])
```

Comment:
Validate that the BodyRevisions property is present and correct, wherever listRevisions=true is set.

Reply:
Changed are committed to validate that the BodyRevisions property is present and correct, wherever listRevisions=true is set.